apps/src/sites/studio/pages/signup.js

```diff
@@ -115,18 +123,96 @@ window.SignupManager = function (options) {
 
     // Force teachers to explicitly accept terms of service.
     $("#user_terms_of_service_version").prop('checked', false);
+
+    logTeacherToggle(true);
+    lastUserType = "teacher";
+  }
+
+  /**
+   * Log signup-related analytics events to Firehose
+   * @param eventName name of the event to log
+   * @param extraData optional hash object for supplemental data (will show up in the data_json field)
+   */
+  function logAnalyticsEvent(eventName, extraData = {}) {
+    if (!self.uuid) {
+      if (!!window.localStorage && !!window.localStorage.getItem("analyticsID")) {
+        self.uuid = window.localStorage.getItem("analyticsID");
+      } else {
+        self.uuid = createUuid();
+        trySetLocalStorage("analyticsID", self.uuid);
+      }
+    }
+
+    const streamName = "analysis-events";
+    const environment = "eric-dev"; // TODO(eric): switch this to production before merge
+    const study = "signup_school_dropdown";
+    const studyGroup = shouldShowSchoolDropdown() ? "show_school_dropdown" : "control";
+
+    let dataJson = {
+      user_agent: window.navigator.userAgent,
+      window_width: window.innerWidth,
+      window_height: window.innerHeight,
+    };
+    Object.assign(dataJson, extraData);
+    if (!!window.optimizely) {
+      const optimizelyData = {
+        optimizely_data: window.optimizely.data.state
+      };
+      Object.assign(dataJson, optimizelyData);
+    }
+
+    firehoseClient.putRecord(
+      streamName,
+      {
+        created_at: new Date().toISOString(),
+        environment: environment,
+        study: study,
+        study_group: studyGroup,
+        event: eventName,
+        data_string: self.uuid,
```

Comment:
This seems worth creating a separate column for, rather than using `data_string` for that purpose. Assuming you don't disagree, I'll make the requisite infrastructure changes.